specs/src/tx_client_v2.md

```diff
@@ -0,0 +1,148 @@
+# Tx Client v2 – Sequential Submission Queue
+
+### Context & Motivation
+
+Tx Client v2 enforces a per-account sequential submission queue that serializes signing, submission, and recovery, avoiding sequence races and minimizing transaction failures.
+
+## Protocol/Component Description
+
+### Per account sequential queue
+
+Every account gets its own **transaction queue**.
+
+This queue is in charge of everything for that account: taking new transactions(jobs), broadcasting them in submission order, reporting outcomes.
+
+Inside the queue we keep two kinds of transactions:
+
+- **Queued** - submitted but not broadcast yet
+- **Pending** - already broadcast and waiting for confirmation
+
+---
+
+### Broadcast and Confirmation loops
+
+Each queue runs two background loops:
```

Comment:
I would think that the number of loops should be bounded to some fixed number, because the number of signers can be arbitrary high.

E.g. we can just have let's say X (fixed number) of workers which just perform the tasks (e.g. we can have X submission workers and X confirmation workers). This will allow enough parallelism without creating too much contention.

Reply:
> I would think that the number of loops should be bounded to some fixed number, because the number of signers can be arbitrary high.

I doubt that number of signers actively submitting transactions simultaniously would be arbitrarily high but it doesn't hurt to cap number of workers anyways